### DISCUSSION TITLE
How to use custom transforms for training and evaluation

### DISCUSSION THREAD

**Discussion by letmejoin** [open-edge-platform/anomalib#2086] [Q&A] [answered]
### What is the motivation for this task?

1. new version anomalib reduce transforms from dataset Calss, but I don't konw how to add it with config.yaml
2. The error message do not show enough information for solving
3. ![image](https://github.com/openvinotoolkit/anomalib/assets/24537107/8f44eb10-9017-4880-ad1e-c66eff428e65)

### Describe the solution you'd like

Pleace give usage for customed data preprocessing, such as CenterCrop, Normalize

**Comment by samet-akcay** [comment]
Can you share your config file here?

**Reply by letmejoin** [reply]
That's my config,thanks!
seed_everything: true
trainer:
  accelerator: auto
  strategy: auto
  devices: 1
  num_nodes: 1
  precision: 32
  logger: null
  callbacks: null
  fast_dev_run: false
  max_epochs: null
  min_epochs: null
  max_steps: -1
  min_steps: null
  max_time: null
  limit_train_batches: null
  limit_val_batches: null
  limit_test_batches: null
  limit_predict_batches: null
  overfit_batches: 0.0
  val_check_interval: null
  check_val_every_n_epoch: 1
  num_sanity_val_steps: null
  log_every_n_steps: null
  enable_checkpointing: null
  enable_progress_bar: null
  enable_model_summary: null
  accumulate_grad_batches: 1
  gradient_clip_val: null
  gradient_clip_algorithm: null
  deterministic: null
  benchmark: null
  inference_mode: true
  use_distributed_sampler: true
  profiler: null
  detect_anomaly: false
  barebones: false
  plugins: null
  sync_batchnorm: true
  reload_dataloaders_every_n_epochs: 0
  default_root_dir: patchcore_train
normalization:
  normalization_method: MIN_MAX

task: CLASSIFICATION
metrics:
  image:
  - AUROC
  - F1Score
  pixel: null
  threshold:
    class_path: anomalib.metrics.F1AdaptiveThreshold
    init_args:
      default_value: 0.5
      thresholds: null
      validate_args: true
      compute_on_cpu: false
      dist_sync_on_step: false
      sync_on_compute: true
logging:
  log_graph: false
default_root_dir: 'resutls/diamond_andriod'
ckpt_path: null
data:
  class_path: anomalib.data.Folder
  init_args:
    name: iphone
    normal_dir: train/good
    root: datasets/Diamond_andriod
    abnormal_dir: test/bad
    normal_test_dir: test/good
    mask_dir: null
    normal_split_ratio: 0.0
    extensions: [".jpeg", ".jpg"]
    image_size:
    - 1280 # dimensions to which images are resized (mandatory)
    - 1280
    train_batch_size: 4
    eval_batch_size: 1
    num_workers: 10
    task: CLASSIFICATION
    transform: 
      class_path: torchvision.transforms.v2.Normalize
      init_args:
        mean: [0.485, 0.456, 0.406]
        std: [0.229, 0.224, 0.225]
      class_path: torchvision.transforms.v2.CenterCrop
      init_args:
        size: 
        - 448
        - 448
    test_split_mode: from_dir
    test_split_ratio: 0.2
    val_split_mode: same_as_test
    val_split_ratio: 0.5
    seed: null

model:
  class_path: anomalib.models.Patchcore
  init_args:
    backbone: tf_efficientnet_b6_ns
    layers:
    - blocks.2.5
    - blocks.4.7
    pre_trained: true
    coreset_sampling_ratio: 0.1
    num_neighbors: 9

**the rest of my config which commented:
 visualization:
   class_path: BaseVisualizer
   visualizers:
     class_path: ImageVisualizer
     init_args:
       mode: FULL
      task: CLASSIFICATION
   save: True
   log: True
   show: False**

**Comment by samet-akcay** [comment, marked as answer]
This documentation shows how to enable custom transforms via both API and CLI
https://anomalib.readthedocs.io/en/latest/markdown/guides/how_to/data/transforms.html

**Reply by letmejoin** [reply]
Thanks for your replay！I get it!
But for other usage, I still don't know how to setup. 
For example, how to modify the saving path of inference result images? The current image saving path is nested very deeply; the configuration parameters of visualization and visualizers seem to have been removed from the previous version; the specific code of the export model is also missing. How to configure it? The parameters of the exported model are not introduced in detail.
I feel that a detailed documentation is needed for each configuration parameter and configuration example of the configuration parameter, just like the configuration example of transform you gave me, but I didn't know where to find this introduction before.
I hope you can understand my explanation and be more friendly to developers who are new to this platform. Thank you very much!
